File: hailiao_flutter/lib/models/report_dto.dart
class ReportDTO {
  final int? id;
  final int? reporterId;
  final int? targetId;
  final int? targetType;
  final String? targetTypeLabel;
  final String? reason;
  final String? evidence;
  final int? status;
  final String? statusLabel;
  final int? handlerId;
  final String? handleResult;
  final String? createdAt;
  final String? handledAt;

  ReportDTO({
    this.id,
    this.reporterId,
    this.targetId,
    this.targetType,
    this.targetTypeLabel,
    this.reason,
    this.evidence,
    this.status,
    this.statusLabel,
    this.handlerId,
    this.handleResult,
    this.createdAt,
    this.handledAt,
  });

  factory ReportDTO.fromJson(Map<String, dynamic> json) {
    return ReportDTO(
      id: (json['id'] as num?)?.toInt(),
      reporterId: (json['reporterId'] as num?)?.toInt(),
      targetId: (json['targetId'] as num?)?.toInt(),
      targetType: (json['targetType'] as num?)?.toInt(),
      targetTypeLabel: json['targetTypeLabel'] as String?,
      reason: json['reason'] as String?,
      evidence: json['evidence'] as String?,
      status: (json['status'] as num?)?.toInt(),
      statusLabel: json['statusLabel'] as String?,
      handlerId: (json['handlerId'] as num?)?.toInt(),
      handleResult: json['handleResult'] as String?,
      createdAt: json['createdAt'] as String?,
      handledAt: json['handledAt'] as String?,
    );
  }
}

File: hailiao-api/src/main/java/com/hailiao/api/dto/ReportDTO.java
package com.hailiao.api.dto;

import java.util.Date;

public class ReportDTO {
    private Long id;
    private Long reporterId;
    private Long targetId;
    private Integer targetType;
    private String targetTypeLabel;
    private String reason;
    private String evidence;
    private Integer status;
    private String statusLabel;
    private Long handlerId;
    private String handleResult;
    private Date createdAt;
    private Date handledAt;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Long getReporterId() { return reporterId; }
    public void setReporterId(Long reporterId) { this.reporterId = reporterId; }

    public Long getTargetId() { return targetId; }
    public void setTargetId(Long targetId) { this.targetId = targetId; }

    public Integer getTargetType() { return targetType; }
    public void setTargetType(Integer targetType) { this.targetType = targetType; }

    public String getTargetTypeLabel() { return targetTypeLabel; }
    public void setTargetTypeLabel(String targetTypeLabel) { this.targetTypeLabel = targetTypeLabel; }

    public String getReason() { return reason; }
    public void setReason(String reason) { this.reason = reason; }

    public String getEvidence() { return evidence; }
    public void setEvidence(String evidence) { this.evidence = evidence; }

    public Integer getStatus() { return status; }
    public void setStatus(Integer status) { this.status = status; }

    public String getStatusLabel() { return statusLabel; }
    public void setStatusLabel(String statusLabel) { this.statusLabel = statusLabel; }

    public Long getHandlerId() { return handlerId; }
    public void setHandlerId(Long handlerId) { this.handlerId = handlerId; }

    public String getHandleResult() { return handleResult; }
    public void setHandleResult(String handleResult) { this.handleResult = handleResult; }

    public Date getCreatedAt() { return createdAt; }
    public void setCreatedAt(Date createdAt) { this.createdAt = createdAt; }

    public Date getHandledAt() { return handledAt; }
    public void setHandledAt(Date handledAt) { this.handledAt = handledAt; }
}
